src/main/java/com/mongodb/jdbc/MongoJsonSchema.java

```diff
@@ -164,6 +176,78 @@ private static MongoJsonSchema toMongoJsonSchema(JsonSchema baseSchema) {
         return result;
     }
 
+    /**
+     * Converts a polymorphic items field which can either be a JsonSchema or an Array of JsonSchema
+     * to a set of MongoJsonSchema.
+     *
+     * @param polymorphicItems The original polymorphic field.
+     * @return the corresponding MongoJsonSchema set.
+     */
+    private static Set<MongoJsonSchema> polymorphicItemsToMongoJsonSchemaSet(
+            BsonValue polymorphicItems) {
+        if (polymorphicItems == null) {
+            return null;
+        }
+
+        // The only expected types for Items are BsonArray or BsonDocument
+        if (!(polymorphicItems.isArray() || polymorphicItems.isDocument())) {
+            throw new BsonInvalidOperationException(
+                    "Value expected to be of type "
+                            + BsonType.ARRAY
+                            + " or "
+                            + BsonType.DOCUMENT
+                            + " but  is of unexpected type "
+                            + polymorphicItems.getBsonType());
+        }
+
+        BsonArray items =
+                polymorphicItems.isDocument()
+                        ? new BsonArray(Collections.singletonList(polymorphicItems.asDocument()))
+                        : polymorphicItems.asArray();
+
+        return items.asArray()
+                .stream()
+                .map(
+                        val ->
+                                toSimplifiedMongoJsonSchema(
+                                        JSON_SCHEMA_CODEC.decode(
+                                                new JsonReader(val.asDocument().toJson()),
+                                                DecoderContext.builder().build())))
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Converts a polymorphic additionalProperties which can either be a boolean or a Document to a
+     * boolean.
+     *
+     * @param polymorphocAdditionalProperties The original polymorphic additionalProperties field.
+     * @return the corresponding boolean value.
+     */
+    private static boolean toMongoJsonSchemaAdditionalProperties(
+            BsonValue polymorphocAdditionalProperties) {
+        if (polymorphocAdditionalProperties == null) {
+            // By default, additional properties is false
```

Comment:
In the [document](https://docs.mongodb.com/manual/reference/operator/query/jsonSchema/) linked in the comments, the behavior description says additionalProperties defaults to `true`.  Is this defaulting to false specific to the JDBC behavior?

Reply:
umh, this was the original behavior, no change there. I'm not sure why it was like that.

Reply:
Taking this back. If no value is provided, it seems to make more sense to default to false, no?
This is also what isAny expects.

Reply:
Yes, let's keep it default to false as that is the current behavior.